packages/xrpl/src/sugar/submit.ts

```diff
@@ -161,14 +61,49 @@ async function submitRequest(
   return client.request(request)
 }
 
-/*
- * The core logic of reliable submission.  This polls the ledger until the result of the
- * transaction can be considered final, meaning it has either been included in a
- * validated ledger, or the transaction's lastLedgerSequence has been surpassed by the
- * latest ledger sequence (meaning it will never be included in a validated ledger).
+/**
+ * Waits for the final outcome of a transaction by polling the ledger until the result can be considered final,
+ * meaning it has either been included in a validated ledger, or the transaction's lastLedgerSequence has been
+ * surpassed by the latest ledger sequence (meaning it will never be included in a validated ledger).
```

Comment:
If we're exporting this, I think it's worth including in this description that likely the function people will want to use is `submit` or `submitAndWait` instead of this one. The use for this is very niche in my head, and I'm worried someone will accidentally use it if VSCode gives them code autocomplete and then get stuck if they don't know about the more generally applicable functions. 

I may suggest keeping it private to avoid the confusion altogether since I don't see when someone would want to use this function instead of `submitAndWait`.

Reply:
It doesn't show up publicly. This file isn't exported for people to render in the docs.

Reply:
Ah I didn't realize that :)